src/librustc/session/config.rs

```diff
@@ -2344,10 +2344,10 @@ fn parse_opt_level(
         }
     };
 
-    // Disable hardware tracing when optimisations are enabled.
-    // https://github.com/softdevteam/ykrustc/issues/66
-    if opt_level != OptLevel::No && cg.tracer == TracerMode::Hardware {
-       cg.tracer = TracerMode::Software;
+    // Currently we disallow optimisation *and* tracing, as we don't want LLVM to re-order things
```

Comment:
This doesn't read well: I think we want to say something like "We don't support both optimisation and tracing being enabled simultaneously as LLVM is likely to reorder things, destroying the correctness of our labels".

Reply:
How's my last push?